File: sdk/samples/all in on code/Visual Studio 2008/VBWinFormDesigner/CustomEditors/EditorForm.vb
'************************************* Module Header **************************************\
' Module Name:	EditorForm.vb
' Project:		VBWinFormDesigner
' Copyright (c) Microsoft Corporation.
' 
' 
' The CustomEditor sample demonstrates how to use a custom editor for a specific property 
' at design time. 
' 
' This source is subject to the Microsoft Public License.
' See http://www.microsoft.com/opensource/licenses.mspx#Ms-PL.
' All other rights reserved.

' THIS CODE AND INFORMATION IS PROVIDED "AS IS" WITHOUT WARRANTY OF ANY KIND,
' EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE IMPLIED
' WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A PARTICULAR PURPOSE.
'***************************************************************************'

Namespace CustomEditors
    Public Class EditorForm
        Private _subCls As SubClass = New SubClass

        Public Sub New()
            Me.InitializeComponent()
            AddHandler MyBase.Load, New EventHandler(AddressOf Me.EditorForm_Load)
        End Sub

        Private Sub button1_Click(ByVal sender As Object, ByVal e As EventArgs)
            MyBase.DialogResult = DialogResult.OK
        End Sub

        Private Sub button2_Click(ByVal sender As Object, ByVal e As EventArgs)
            MyBase.DialogResult = DialogResult.Cancel
        End Sub

        Private Sub EditorForm_Load(ByVal sender As Object, ByVal e As EventArgs)
            Me.textBox1.Text = Me._subCls.Name
            Me.monthCalendar1.SelectionStart = Me._subCls.MyDate
            Me.monthCalendar1.SelectionEnd = Me._subCls.MyDate
        End Sub

        Public Property SubCls() As SubClass
            Get
                Me._subCls.Name = Me.textBox1.Text
                Me._subCls.MyDate = Me.monthCalendar1.SelectionStart
                Return Me._subCls
            End Get
            Set(ByVal value As SubClass)
                Me._subCls = value
            End Set
        End Property

    End Class
End Namespace

File: sdk/samples/all in on code/Visual Studio 2008/CSWinFormDesigner/CustomEditors/EditorForm.cs
/************************************* Module Header **************************************\
* Module Name:	EditorForm.cs
* Project:		CSWinFormDesigner
* Copyright (c) Microsoft Corporation.
* 
* 
* The CustomEditor sample demonstrates how to use a custom editor for a specific property 
* at design time. 
* 
* This source is subject to the Microsoft Public License.
* See http://www.microsoft.com/opensource/licenses.mspx#Ms-PL.
* All other rights reserved.
* 
* History:
* * 5/25/2009 3:00 PM Zhi-Xin Ye Created
* * 
* 
\******************************************************************************************/

using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Windows.Forms;

namespace CSWinFormDesigner.CustomEditors
{
    public partial class EditorForm : Form
    {
        public EditorForm()
        {
            InitializeComponent();
            this.Load += new EventHandler(EditorForm_Load);
        }

        void EditorForm_Load(object sender, EventArgs e)
        {
            this.textBox1.Text = this.subCls.Name;
            this.monthCalendar1.SelectionStart = this.subCls.Date;
            this.monthCalendar1.SelectionEnd = this.subCls.Date;
        }
        private SubClass subCls = new SubClass();

        public SubClass SubCls
        {
            get
            {
                this.subCls.Name = this.textBox1.Text;
                this.subCls.Date = this.monthCalendar1.SelectionStart;
                return subCls;
            }
            set
            {
                this.subCls = value;
            }
        }

        private void button1_Click(object sender, EventArgs e)
        {
            this.DialogResult = DialogResult.OK;

        }

        private void button2_Click(object sender, EventArgs e)
        {
            this.DialogResult = DialogResult.Cancel;
        }
    }
}
